src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1779,7 +1818,15 @@ private void WriteRedFactory(Node nd)
                     Write(", ");
                 if (field.Type == "SyntaxToken")
                 {
-                    Write("(Syntax.InternalSyntax.SyntaxToken){0}.Node", CamelCase(field.Name));
+                    if (IsOptional(field))
+                    {
+                        Write("(Syntax.InternalSyntax.SyntaxToken?){0}.Node", CamelCase(field.Name));
+                    }
+                    else
+                    {
+                        // We know the GreenNode is not null because it gets a type check earlier in the generated method
+                        Write("(Syntax.InternalSyntax.SyntaxToken){0}.Node!", CamelCase(field.Name));
```

Comment:
What warning is this suppressing? `{0}` was indeed null checked, and accessing should not be warning.

Reply:
`SyntaxToken.Node` is nullable. The type isn't annotated yet, but it will be shortly.